packages/components/src/border-control/types.ts

```diff
@@ -0,0 +1,160 @@
+export type Border = {
+	color?: string;
+	style?: string;
+	width?: string;
```

Comment:
Have we considered using `CSSProperties` to type these props?

Similar question for the `Color` type just below.

Reply:
It wasn't high on my list of considerations. 

I can see `CSSProperties` being more descriptive and restrictive than a simple `string` type. How would we best express that the values accepted are only a subset of the `CSSProperties`? For example I don't think we'll want to allow `clamp(x, y, z)` or `var(--anything)` etc.

Reply:
> How would we best express that the values accepted are only a subset of the CSSProperties? For example I don't think we'll want to allow clamp(x, y, z) or var(--anything) etc.

Mhh, not sure if we'd be able to do that without adding it explicitly in the prop's docs AND potentially performing a runtime check.

We'd have the same problem with `string` too, though — and that's why I think that potentially using `CSSProperties[ 'width' ]` may still be a better option here, as it shows the purpose of this prop a bit better.

Reply:
I've trialled switching these types to `CSSProperties`, for example, `width` and the more appropriate `borderWidth` however I end up with additional type errors to solve. One issue is `CSSProperties['width']` now allowing `number` values.

Ultimately, I was able to resolve the typing errors but it was forcing a rewrite of valid function code just to satisfy the typing. This seems wrong and undesirable to me so I haven't pushed the changes. I'll include a patch below that leaves only the final typing error I'm referring to here. 

Also, for the record, I abandoned this change before retesting the components still function correctly (isolated here and in their real world uses where the input value could be seeded via theme.json). I'm not convinced allowing number values for the widths is desirable.

<details>
<summary> Example changes </summary>

```diff
diff --git a/packages/components/src/border-box-control/types.ts b/packages/components/src/border-box-control/types.ts
index 1522228358..68790bc94f 100644
--- a/packages/components/src/border-box-control/types.ts
+++ b/packages/components/src/border-box-control/types.ts
@@ -11,8 +11,8 @@ export type Borders = {
 };
 
 export type AnyBorder = Border | Borders | undefined;
-export type BorderProp = 'color' | 'style' | 'width';
-export type BorderSide = 'top' | 'right' | 'bottom' | 'left';
+export type BorderProp = keyof Border;
+export type BorderSide = keyof Borders;
 
 export type BorderBoxControlProps = {
 	/**
diff --git a/packages/components/src/border-box-control/utils.ts b/packages/components/src/border-box-control/utils.ts
index 67ce4d8a39..46515784b6 100644
--- a/packages/components/src/border-box-control/utils.ts
+++ b/packages/components/src/border-box-control/utils.ts
@@ -1,3 +1,8 @@
+/**
+ * External dependencies
+ */
+import type { CSSProperties } from 'react';
+
 /**
  * Internal dependencies
  */
@@ -94,9 +99,9 @@ export const getCommonBorder = ( borders?: Borders ) => {
 		return undefined;
 	}
 
-	const colors: ( string | undefined )[] = [];
-	const styles: ( string | undefined )[] = [];
-	const widths: ( string | undefined )[] = [];
+	const colors: ( CSSProperties[ 'borderColor' ] | undefined )[] = [];
+	const styles: ( CSSProperties[ 'borderStyle' ] | undefined )[] = [];
+	const widths: ( CSSProperties[ 'borderWidth' ] | undefined )[] = [];
 
 	sides.forEach( ( side ) => {
 		colors.push( borders[ side ]?.color );
diff --git a/packages/components/src/border-control/types.ts b/packages/components/src/border-control/types.ts
index 8834e0cc94..cdae72ffb4 100644
--- a/packages/components/src/border-control/types.ts
+++ b/packages/components/src/border-control/types.ts
@@ -1,12 +1,17 @@
+/**
+ * External dependencies
+ */
+import type { CSSProperties } from 'react';
+
 export type Border = {
-	color?: string;
-	style?: string;
-	width?: string;
+	color?: CSSProperties[ 'borderColor' ];
+	style?: CSSProperties[ 'borderStyle' ];
+	width?: CSSProperties[ 'borderWidth' ];
 };
 
 export type Color = {
 	name: string;
-	color: string;
+	color: CSSProperties[ 'color' ];
 };
 
 export type BorderControlProps = {

```
</details>

Feel free to take a look and let me know where I've gone wrong.

Reply:
In the interests of moving this forward, I've refactored the `getBorderDiff` util function so that TypeScript won't complain about looping the border properties.